### PR TITLE
Fix overloaded tx

### PR DESCRIPTION
**Pull request by AloeareV**
This should be clearer. Big change.

### CODE REVIEW

**Comment by zancas** [comment]
I will check this out and see if the tests run.

**Comment by dannasessha** [comment]
I'm digging into this, the plan is to make two passes.

The first will be to assure that name changes have occurred properly (ie, the code's functioning is unchanged), and the second will be to examine the names themselves (ie, do the names make consistent sense).

I'll also make some notes with `cargo tarpaulin` here.

**Comment by dannasessha** [comment]
Tests are passing on both `dev` and `fix_overloaded_tx` branches. 

Running `cargo tarpaulin --avoid-cfg-tarpaulin` against...
`dev` shows `63.37% coverage, 3991/6298 lines covered`
`fix_overloaded_tx` shows `63.62% coverage, 4051/6367 lines covered`

File: cli/src/lib.rs [hunk 1]
```diff
@@ -113,20 +113,20 @@ pub fn startup(
     }
 
     // Start the command loop
-    let (command_tx, resp_rx) = command_loop(lightclient.clone());
+    let (command_transmitter, resp_receiver) = command_loop(lightclient.clone());
```
Comment (zancas) [inline cli/src/lib.rs:116]: @dannasessha to sanity check this I will check the return type of a `command_loop`.

If the returned type matches the new ident, then that's sufficient for me to sign off.

Comment (zancas) [reply cli/src/lib.rs:116]: The return type uses is "sender" but...   we can swap `transmitter` for `sender` some other time...   or we could do it now.. 

I'm agnostic.

Comment (zancas) [reply cli/src/lib.rs:116]: Oh...  I send `transmitter` sounds less like something that distributes a `transaction`?

**Comment by zancas** [comment]
@dannasessha I am signing off on:

1. [cli/src/lib.rs](https://github.com/zingolabs/zecwallet-light-cli/pull/12/files#diff-c1f8f7498da827a634bddc8a7559198bc99b296e9d9e8b91a70b503662995b8c)
2. [cli/src/main.rs](https://github.com/zingolabs/zecwallet-light-cli/pull/12/files#diff-37970e7d817d4316a093bb967d74ff7e26d9c6c01d93f5a79d3d195ba9f6155b)

I reviewed them, and think they are correct.

I am now reviewing:

[lib/proto/compact_formats.proto](https://github.com/zingolabs/zecwallet-light-cli/pull/12/files#diff-22dff1068518f65b35cebe8e1581f049ffd72a978500ce62069e807618912749)

In the interest of efficiency you should focus on a different part of the diff.

File: lib/proto/service.proto [hunk 2]
```diff
@@ -22,10 +22,10 @@ message BlockRange {
     BlockID end = 2;
 }
 
-// A TxFilter contains the information needed to identify a particular
+// A TransactionFilter contains the information needed to identify a particular
```
Comment (zancas) [inline lib/proto/service.proto:25]: All these substitutions are correct in the sense that they are indeed naming transaction concepts, but do we need to be concerned about compatibility with the `lightwalletd` gRPC interface?  I think @dannasessha raised this issue?

File: lib/src/blaze/block_witness_data.rs [hunk 3]
```diff
@@ -1,13 +1,13 @@
 use crate::{
-    compact_formats::{CompactBlock, CompactTx, TreeState},
+    compact_formats::{CompactBlock, CompactTransaction, TreeState},
```
Comment (zancas) [inline lib/src/blaze/block_witness_data.rs:2]: I see that this is the correct name given the definition in the `compact_formats.proto`..   my concern is that I don't know if it's interoperable with the `lightwalletd` server.

File: lib/src/blaze/block_witness_data.rs [hunk 4]
```diff
@@ -256,35 +256,35 @@ impl BlockAndWitnessData {
...
 
     /// Start a new sync where we ingest all the blocks
     pub async fn start(
         &self,
         start_block: u64,
         end_block: u64,
-        wallet_txns: Arc<RwLock<WalletTxns>>,
-        reorg_tx: UnboundedSender<Option<u64>>,
+        wallet_transactions: Arc<RwLock<WalletTxns>>,
+        reorg_transmitter: UnboundedSender<Option<u64>>,
```
Comment (zancas) [inline lib/src/blaze/block_witness_data.rs:281]: Yikes!  There's an adjacency!

File: lib/src/blaze/block_witness_data.rs [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
+    compact_formats::{CompactBlock, CompactTransaction, TreeState},
     grpc_connector::GrpcConnector,
     lightclient::{
         checkpoints::get_all_main_checkpoints,
         lightclient_config::{LightClientConfig, MAX_REORG},
     },
     lightwallet::{
         data::{BlockData, WalletTx, WitnessCache},
-        wallet_txns::WalletTxns,
+        wallet_transactions::WalletTxns,
```
Comment (dannasessha) [inline lib/src/blaze/block_witness_data.rs:10]: `WalletTxns` unchanged here

File: lib/src/blaze/block_witness_data.rs [hunk 6]
```diff
@@ -105,7 +105,7 @@ impl BlockAndWitnessData {
         }
     }
 
-    pub async fn get_ctx_for_nf_at_height(&self, nullifier: &Nullifier, height: u64) -> (CompactTx, u32) {
+    pub async fn get_ctx_for_nf_at_height(&self, nullifier: &Nullifier, height: u64) -> (CompactTransaction, u32) {
```
Comment (dannasessha) [inline lib/src/blaze/block_witness_data.rs:108]: What a fn name! Might be a good one to look at, while we're at it?

File: lib/src/blaze/block_witness_data.rs (hunk 4 above)
Comment (dannasessha) [inline lib/src/blaze/block_witness_data.rs:281]: `reorg_transmitter` is of type `UnboundedSender<Option<u64>>`
... maybe that's not bad! But what about calling it a sender, possibly?

File: lib/src/blaze/block_witness_data.rs [hunk 7]
```diff
@@ -468,12 +469,12 @@ impl BlockAndWitnessData {
         };
 
         // Go over all the outputs. Remember that all the numbers are inclusive, i.e., we have to scan upto and including
-        // block_height, tx_num and output_num
-        for (t_num, ctx) in cb.vtx.iter().enumerate() {
-            for (o_num, co) in ctx.outputs.iter().enumerate() {
+        // block_height, transaction_num and output_num
```
Comment (dannasessha) [inline lib/src/blaze/block_witness_data.rs:472]: `tx_num` changed to `transaction_num` here (in comments)

File: lib/src/blaze/block_witness_data.rs [hunk 8]
```diff
@@ -468,12 +469,12 @@ impl BlockAndWitnessData {
...
+        for (t_num, compact_transaction) in cb.v_transaction.iter().enumerate() {
```
Comment (dannasessha) [inline lib/src/blaze/block_witness_data.rs:473]: ... `t_num` here unchanged. Looks to me like the comment is referring to  this variable.

**Comment by dannasessha** [comment]
> @dannasessha I am signing off on:
>     1. [cli/src/lib.rs](https://github.com/zingolabs/zecwallet-light-cli/pull/12/files#diff-c1f8f7498da827a634bddc8a7559198bc99b296e9d9e8b91a70b503662995b8c)
>     2. [cli/src/main.rs](https://github.com/zingolabs/zecwallet-light-cli/pull/12/files#diff-37970e7d817d4316a093bb967d74ff7e26d9c6c01d93f5a79d3d195ba9f6155b)
>     3. [lib/proto/compact_formats.proto](https://github.com/zingolabs/zecwallet-light-cli/pull/12/files#diff-22dff1068518f65b35cebe8e1581f049ffd72a978500ce62069e807618912749)
>     4. [lib/proto/service.proto](https://github.com/zingolabs/zecwallet-light-cli/pull/12/files#diff-c1b60628247db92a4cd4a6dd7559643d1725651fef66f51069103a650685ccc5)
>     5. [lib/src/blaze.rs](https://github.com/zingolabs/zecwallet-light-cli/pull/12/files#diff-5201d3a90bca7fdd3dd1115c50941b831d2c68f0e5ae95e2de2a009b9b120e65)
>     6. [lib/src/blaze/block_witness_data.rs](https://github.com/zingolabs/zecwallet-light-cli/pull/12/files#diff-fe24446938f05b862f0ff1d05aa869ec72d727a008b5c335db2cb868768b8a68)

> I reviewed them, and think they are internally correct. To merge this code we will need to validate that interfacing projects still interoperate.

I agree with these, though I didn't finish reviewing the `block_witness_data.rs` file, and I didn't have LSP support for the `proto` files.

> Consumers/Interactors.:
> 
>     1. the cli                                                                 (this repo)
> 
>     2. https://github.com/zingolabs/zecwallet-mobile  (android and iOS)
> 
>     3. https://github.com/zingolabs/zecwallet-lite        (the desktop)
> 
>     4. https://github.com/zingolabs/lightwalletd           (the server)
> 
> 
> In the interest of efficiency you should focus on a different part of the diff.

I also agree we need to consider these. Is there any chance there are more consumers outside our group of codebases you've listed here?

**Comment by dannasessha** [comment]
Just a soft suggestion: could we break this up into chunks, might that help?

File: lib/src/blaze/block_witness_data.rs (hunk 4 above)
Comment (AloeareV) [reply lib/src/blaze/block_witness_data.rs:281]: I believe all `transmitter`s are of some `Sender` type.

**Comment by zancas** [comment]
You could open new PRs that contain well encapsulated chunks of this work, if you see how to.

File: lib/src/blaze/block_witness_data.rs (hunk 5 above)
Comment (zancas) [reply lib/src/blaze/block_witness_data.rs:10]: What do you think of this @AloeareV ?

Comment (AloeareV) [reply lib/src/blaze/block_witness_data.rs:10]: I missed that I hadn't responded to this. It's a public type: 
https://github.com/zingolabs/zecwalletlitelib/blob/b44226dbfb98963dacaf2eac08405b2879c1f988/lib/src/lightwallet/data.rs#L586

**Review by zancas — APPROVED** [review]

File: lib/src/blaze/block_witness_data.rs (hunk 3 above)
Comment (zancas) [reply lib/src/blaze/block_witness_data.rs:2]: We'll discover this with smoke tests.

File: lib/proto/service.proto (hunk 2 above)
Comment (zancas) [reply lib/proto/service.proto:25]: Will smoke.

**Pull request merged**